amplipi/eeprom.py

```diff
@@ -76,166 +140,128 @@ class UnsupportedFormatError(Exception):
 class EEPROM:
   """Class for reading from and writing to EEPROM."""
 
-  def __init__(self, bus: int, board: BoardType) -> None:
-    self._i2c = smbus.SMBus(bus)
-    self._address = board.value
-    try:
-      GPIO.setmode(GPIO.BCM)
-      GPIO.setup(WP_PIN, GPIO.OUT)
-    except NameError:
-      pass
+  EEPROM_PAGE_SIZE: int = 16  # Number of bytes per EEPROM page.
 
-  @staticmethod
-  def get_available_devices(bus: int) -> List[BoardType]:
-    """Get list of available devices."""
-    i2c = smbus.SMBus(bus)
+  def __init__(self, board: BoardType) -> None:
+    self._board = board.value
 
-    devices = []
+  def _bus(self) -> int:
+    """Returns the I2C bus to communicate on, based on the BoardType"""
+    # TODO: make board a full class to abstract this away.
+    return (int(self._board) & 0x80) >> 7  # Bit 7 is bus, either 0 (Pi bus) or 1 (Preamp bus).
 
-    for i in BoardType:
-      try:
-        i2c.read_byte(i.value)
-        devices.append(i)
-      except OSError:
-        pass
-
-    return devices
+  def _address(self) -> int:
+    """Returns the I2C address of the EEPROM, based on the BoardType"""
+    # TODO: make board a full class to abstract this away.
+    return int(self._board) & 0x7F  # Lowest 7 bits are the I2C address
 
+  # TODO: Is this necessary?
   def _enable_write(self) -> None:
-    """Enable write to EEPROM."""
+    """Enable write to EEPROM. Only required for streamers."""
     try:
+      GPIO.setmode(GPIO.BCM)
+      GPIO.setup(WP_PIN, GPIO.OUT)
       GPIO.output(WP_PIN, GPIO.LOW)
     except NameError:
       pass
```

Comment:
Since this should only ever be done on the Streamers, how about
```suggestion
  if self._board == BoardType.STREAMER_SUPPORT:
      try:
        GPIO.setmode(GPIO.BCM)
        GPIO.setup(WP_PIN, GPIO.OUT)
        GPIO.output(WP_PIN, GPIO.LOW)
      except NameError:
        pass
```
And then you can remove that check in the `write_board_info()` function.